app.rb

```diff
@@ -14,19 +16,38 @@ def initialize
   end
 
   def list_all_books
-    puts 'list all books'
+    @books.each do |book|
+      puts "Title: #{book.title}, Publisher: #{book.publisher},
+      Cover state: #{book.cover_state}, Date of Publish: #{book.publish_date}"
+    end
   end
 
   def add_a_book
-    puts 'add a book'
-  end
+    puts 'Please write in title: '
+    title = gets.chomp
 
-  def add_a_movie
-    puts 'add a book'
+    puts 'Please write in publisher: '
+    publisher = gets.chomp
+
+    puts 'Please write in cover state ["good"/"bad"]'
+    cover_state = gets.chomp
+
+    puts 'Please write in date of publish [Enter date in format (yyyy-mm-dd)]'
+    publish_date = gets.chomp
+
+    @books.push(Book.new(title, publisher, cover_state, publish_date))
+    puts 'Book is created'
```

Comment:
- [ OPTIONAL] You did a good job, it is wonderful :+1: :+1:. But I suggest putting this code in a separate module file, so you can only call the `method` at the place (e.g.: `@books.push(method_name())` from your created file for ruby best practices.
- [OPTIONAL ] I think it would be better to give clear messages for **inputs** in order to be understandable for users.